### PR TITLE
Errors on nodes

### LINKED ISSUE #8288: Errors on nodes
If node panics or returns a dataflow error, the message should be displayed on a dark-red/red background as in the new design:

image

* In case of panics, the message should be read from "Panic" payload of expression update
* In case of dataflow errors, the message should be read by attaching special "error" visualization.

For details, see GUI1 implementation.

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Closes #8288
  - Show both `Panic`s and `DataflowError`s on nodes
  - Update Code Editor diagnostics to use the same functionality to properly display `DataflowError`s

### Important Notes
- Does not include warnings
- :heavy_check_mark: Done basic testing to ensure that errors *should* disappear and reappear as appropriate

### Screenshots
(Update) Fixed boldness
image
:warning: Does not play nice with visualizations - it covers up the visualization toolbars:
image
image
Original screenshot:
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

File: app/gui2/src/util/computedValueRegistry.ts [hunk 1]
```diff
@@ -23,15 +26,22 @@ class ComputedValueDb extends ReactiveDb<ExpressionId, ExpressionInfo> {
 /** This class holds the computed values that have been received from the language server. */
 export class ComputedValueRegistry {
   public db = new ComputedValueDb()
+  private dataflowErrorVisualizationIds = new Map<ExpressionId, Uuid>()
   private _updateHandler = this.processUpdates.bind(this)
   private executionContext: ExecutionContext | undefined
+  private visualizationDataRegistry: VisualizationDataRegistry | undefined
```
Comment (farmaazon) [inline app/gui2/src/util/computedValueRegistry.ts]: I think this is mixing responsibilities. `ComputedValueRegistry` is about expression updates received from the engine. It should not handle data flow error visualizations. Let's make a separate util/store for it.

**Review by farmaazon — CHANGES_REQUESTED** [review]

File: app/gui2/src/stores/project.ts [hunk 2]
```diff
@@ -519,8 +519,46 @@ export const useProjectStore = defineStore('project', () => {
...
+  function getDataflowError(exprId: ExpressionId) {
+    const id = dataflowErrorVisualizationIds.get(exprId)
+    if (!id) return
+    return shallowRef(
+      computed<{ kind: 'Dataflow'; message: string } | undefined>(() => {
+        const json = visualizationDataRegistry?.getRawData(id)
+        return json != null ? JSON.parse(json) : undefined
+      }),
+    )
+  }
```
Comment (farmaazon) [inline app/gui2/src/stores/project.ts]: Now it's better, but I would still encapsulate `dataflowErrorVisualizationIds` and getDataflowError in a separate class/module :)

**Review by farmaazon — APPROVED** [review]

**Comment by somebody1234** [comment]
@farmaazon the review issue should now be addressed. i don't think the current solution is ideal either - but i'm not sure how else to ensure that there is exactly one dataflow error viz per dataflow error.

the current solution updates the presence/absence of a dataflow error on each update... but of course that requires access to every update - see the new hacky solution.

i did try using reactivity - e.g. `watchEffect` on a `ReactiveIndex`'s `reverseLookup`, but it seems like that doesn't work. @vitvakatu do you have any thoughts on whether this is possible using `ReactiveIndex`? (or perhaps even just the base `ReactiveDb`?

**Comment by somebody1234** [comment]
i've taken the liberty to also move the other registries from `util` into `stores/project` - let me know if they should go somewhere else instead

File: app/gui2/src/stores/project/dataflowErrorRegistry.ts
Comment (Frizi) [inline app/gui2/src/stores/project/dataflowErrorRegistry.ts:1]: Creating a whole new class and registry, as well as custom `update` event emitted from something that so far had just a standard `ReactiveDb` interface seems like an unnecessary complication for this. How about replacing this whole thing with a `dataflowError` computed value inside a `GraphNode` component, and feeding it into `projectStore.useVisualizationData`?

Comment (somebody1234) [reply app/gui2/src/stores/project/dataflowErrorRegistry.ts:1]: the code editor requires access to every dataflow error, all at the same time

Comment (somebody1234) [reply app/gui2/src/stores/project/dataflowErrorRegistry.ts:1]: however, it *is* true that the diagnostics in the code editor aren't *that* important. alternatively there may be a way for the code editor to grab the required data from the `visualizationDataRegistry`.

**Comment by vitvakatu** [comment]
> i did try using reactivity - e.g. watchEffect on a ReactiveIndex's reverseLookup, but it seems like that doesn't work. @vitvakatu do you have any thoughts on whether this is possible using ReactiveIndex? (or perhaps even just the base ReactiveDb?

That should definitely be possible, reverse index is a reactive map and it should trigger reactive update on changes.

File: app/gui2/src/stores/project/dataflowErrorRegistry.ts
Comment (Frizi) [reply app/gui2/src/stores/project/dataflowErrorRegistry.ts:1]: Good point about the diagnostics, I didn't realised you were using it in multiple places. Still, would it be possible to remove the custom `update` event and use a `ReactiveMapping` that calls `useVisualizationData` inside its getter. That should still be way simpler/less code than current `dataflowErrorRegistry`. That way you would be able to reuse the same visualization from multiple places.

Comment (somebody1234) [reply app/gui2/src/stores/project/dataflowErrorRegistry.ts:1]: ah, forgot about `ReactiveMapping`. yeah, i'll look into it

File: app/gui2/src/stores/project/computedValueRegistry.ts [hunk 3]
```diff
@@ -43,7 +54,9 @@ export class ComputedValueRegistry {
   processUpdates(updates: ExpressionUpdate[]) {
     for (const update of updates) {
       const info = this.db.get(update.expressionId)
-      this.db.set(update.expressionId, combineInfo(info, update))
+      const newInfo = combineInfo(info, update)
+      this.emit('update', [update, info, newInfo])
```
Comment (Frizi) [inline app/gui2/src/stores/project/computedValueRegistry.ts]: Is this emit still used? If not, let's remove the whole `ObservableV2` from here, as it's no longer necessary.

File: app/gui2/src/stores/project/index.ts [hunk 4]
```diff
@@ -555,6 +554,22 @@ export const useProjectStore = defineStore('project', () => {
...
+          ref({
+            expressionId: id,
+            visualizationModule: 'Standard.Visualization.Preprocessor',
+            expression: {
+              module: 'Standard.Visualization.Preprocessor',
+              definedOnType: 'Standard.Visualization.Preprocessor',
+              name: 'error_preprocessor',
+            },
+          }),
+        ) as ShallowRef<{ kind: 'Dataflow'; message: string } | undefined>),
```
Comment (Frizi) [inline app/gui2/src/stores/project/index.ts]: That typecast is very sketchy. I'd prefer to validate the output shape with `zod`, so an unexpected API change/bug on the engine or libs side doesn't bite us here. Too many systems involved to just trust it blindly. I'd rather throw a big error in the console when that happens.

File: app/gui2/src/stores/project/computedValueRegistry.ts (hunk 3 above)
Comment (somebody1234) [reply app/gui2/src/stores/project/computedValueRegistry.ts]: fixed.

File: app/gui2/src/util/database/reactiveDb.ts [hunk 5]
```diff
@@ -326,11 +327,15 @@ export class ReactiveMapping<K, V, M> {
...
     this.computed = reactive(map.create())
     db.on('entryAdded', (key, value, onDelete) => {
-      this.computed.set(
-        key,
-        computed(() => indexer(key, value), debugOptions),
-      )
-      onDelete(() => this.computed.delete(key))
+      const scope = effectScope()
+      const mappedValue = scope.run(() =>
+        computed(() => scope.run(() => indexer(key, value)), debugOptions),
```
Comment (Frizi) [inline app/gui2/src/util/database/reactiveDb.ts:332]: I still think that this is somewhat dodgy to have a `scope.run` inside computed value. We will need to figure out a better way to support this usecase (probably a different db mapper type not based on computed). But for now let's merge it as is and check if this approach has any unintended consequences.

**Review by Frizi — APPROVED** [review]

**Pull request merged**